File: Examples/test-suite/r/li_std_deque_runme.R
clargs <- commandArgs(trailing=TRUE)
source(file.path(clargs[1], "unittest.R"))

dyn.load(paste("li_std_deque", .Platform$dynlib.ext, sep=""))
source("li_std_deque.R")
cacheMetaData(1)

# Test constructors for std::deque<int>
intDeque <- IntDeque()
intDeque2 <- IntDeque(3)
intDeque3 <- IntDeque(4, 42)
intDeque4 <- IntDeque(intDeque3)

# Test constructors for std::deque<double>
doubleDeque  <- DoubleDeque()
doubleDeque2 <- DoubleDeque(3)
doubleDeque3 <- DoubleDeque(4, 42.0)
doubleDeque4 <- DoubleDeque(doubleDeque3)

# Test constructors for std::deque<Real>
realDeque  <- RealDeque()
realDeque2 <- RealDeque(3)
realDeque3 <- RealDeque(4, 42.0)
realDeque4 <- RealDeque(realDeque3)

#### unittest(half(testvec), testvec/2)
#### unittest(average(testvec), mean(testvec))
#### ## string vector test
#### vlen <- 13
#### stringvec <- paste(letters[1:vlen], as.character(rnorm(vlen)))
#### unittest(rev(stringvec), RevStringVec(stringvec))

# average() should return the average of all values in a std::deque<int>
IntDeque_push_back(intDeque, 2)
IntDeque_push_back(intDeque, 4)
IntDeque_push_back(intDeque, 6)
avg <- average(intDeque)
unittest(avg, 4.0)

# half() should return a std::deque<float>, where each element is half
# the value of the corresponding element in the input deque<float>.
# The original deque's contents are unchanged.
#
RealDeque_clear(realDeque)
RealDeque_push_back(realDeque, 2.0)
halfDeque = half(realDeque)
unittest(RealDeque_front(halfDeque), 1.0)

#
# halve_in_place() should...
#
halve_in_place(doubleDeque)

q(save="no")

File: Examples/test-suite/lua/li_std_deque_runme.lua
local v=require("li_std_deque")

-- Test constructors for std::deque<int>
local intDeque  = v.IntDeque()
local intDeque2 = v.IntDeque(3)
local intDeque3 = v.IntDeque(4, 42)
local intDeque4 = v.IntDeque(intDeque3)

-- Test constructors for std::deque<double>
local doubleDeque  = v.DoubleDeque()
local doubleDeque2 = v.DoubleDeque(3)
local doubleDeque3 = v.DoubleDeque(4, 42.5)
local doubleDeque4 = v.DoubleDeque(doubleDeque3)

-- Test constructors for std::deque<Real>
local realDeque  = v.RealDeque()
local realDeque2 = v.RealDeque(3)
local realDeque3 = v.RealDeque(4, 42.5)
local realDeque4 = v.RealDeque(realDeque3)

-- average() should return the average of all values in a std::deque<int>
intDeque:push_back(2)
intDeque:push_back(4)
intDeque:push_back(6)
local avg = v.average(intDeque)
assert(avg == 4)
assert(intDeque:size() == 3)

-- half() should return a std::deque<float>, where each element is half
-- the value of the corresponding element in the input deque<float>.
-- The original deque's contents are unchanged.
realDeque:clear()
realDeque:push_back(5.5)
local halfDeque = v.half(realDeque)
assert(halfDeque:getitem(0) == 2.75)
assert(realDeque:size() == 1)

-- halve_in_place() should...
v.halve_in_place(doubleDeque4)
assert(doubleDeque4:getitem(0) == 21.25)
assert(doubleDeque4:getitem(1) == 21.25)
assert(doubleDeque4:getitem(2) == 21.25)
assert(doubleDeque4:getitem(3) == 21.25)
assert(doubleDeque4:size() == 4)
